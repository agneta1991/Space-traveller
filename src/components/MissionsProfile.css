.missions-profile-table {
  width: 40vw;
  border-collapse: collapse;
  border: 1px solid black;
  margin: 20px;
  padding: 10px;
}

h2 {
  margin: 20px;
}

.missions-profile-table th,
.missions-profile-table td {
  border: 1px solid black;
  padding: 10px;
  text-align: left;
  font-size: 1.2rem;
  font-weight: 500;
}

.no-missions-placeholder {
  font-weight: 600;
  background-color: #f0f0f0;
  border-radius: 4px;
  padding: 10px;
}
