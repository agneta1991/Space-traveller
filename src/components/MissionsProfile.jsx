import React from 'react';
import { useSelector } from 'react-redux';
import { selectMappedMissions } from '../redux/missions/missionsSlice';
import './MissionsProfile.css';

function MissionsProfile() {
  const mappedMissions = useSelector(selectMappedMissions);

  // Filter joined missions
  const joinedMissions = mappedMissions.filter((mission) => mission.reserved);

  return (
    <div>
      <h2>My Missions</h2>
      <table className="missions-profile-table">
        <tbody>
          {joinedMissions.map((mission) => (
            <tr key={mission.mission_id}>
              <td>{mission.mission_name}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}

export default MissionsProfile;
