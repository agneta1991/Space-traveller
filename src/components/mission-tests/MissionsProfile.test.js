import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { toBeInTheDocument } from '@testing-library/jest-dom/matchers';
import MissionsProfile from '../MissionsProfile'; // Update the import path accordingly

expect.extend({ toBeInTheDocument });
const mockStore = configureStore([]);

describe('MissionsProfile Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      missions: {
        data: [
          {
            mission_id: '1',
            mission_name: 'Test Mission 1',
            description: 'Test Description 1',
            reserved: false,
          },
          {
            mission_id: '2',
            mission_name: 'Test Mission 2',
            description: 'Test Description 2',
            reserved: true,
          },
        ],
        status: 'succeeded',
        error: null,
      },
    });
  });

  it('renders joined missions', () => {
    const { container } = render(
      <Provider store={store}>
        <MissionsProfile />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders "No Missions Added" when no missions are joined', () => {
    store = mockStore({
      missions: {
        data: [
          {
            mission_id: '1',
            mission_name: 'Test Mission 1',
            description: 'Test Description 1',
            reserved: false,
          },
        ],
        status: 'succeeded',
        error: null,
      },
    });

    const { getByText, container } = render(
      <Provider store={store}>
        <MissionsProfile />
      </Provider>,
    );

    expect(getByText('No Missions Added')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
