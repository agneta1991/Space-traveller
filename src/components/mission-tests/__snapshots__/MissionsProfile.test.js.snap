// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsProfile Component renders "No Missions Added" when no missions are joined 1`] = `
<div>
  <div>
    <h2>
      My Missions
    </h2>
    <table
      class="missions-profile-table"
    >
      <tbody>
        <tr>
          <td
            class="no-missions-placeholder"
          >
            No Missions Added
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`MissionsProfile Component renders joined missions 1`] = `
<div>
  <div>
    <h2>
      My Missions
    </h2>
    <table
      class="missions-profile-table"
    >
      <tbody>
        <tr>
          <td>
            Test Mission 2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
